WordPress/Classes/ViewRelated/Gutenberg/GutenbergMediaInserterHelper.swift

```diff
@@ -164,6 +178,28 @@ class GutenbergMediaInserterHelper: NSObject {
             gutenberg.mediaUploadUpdate(id: mediaUploadID, state: .uploading, progress: Float(value), url: nil, serverID: nil)
         }
     }
+
+    func fetchVideoPressUrl(for media: Media, completion: @escaping ( Result<(videoURL: URL, posterURL: URL?)> ) -> Void) {
+        guard let videoPressID = media.videopressGUID else {
+            completion(Result.error(NSError()))
```

Comment:
If there is no videoPressID the site can be a self-hosted site without VideoPress so you should return directly the `media.remoteURL`

Reply:
Great insights @SergioEstevao ! Luckily I have a self-hosted site which I only use for testing, I had the chance to test on that and yes media.remoteURL works like a charm on that

Reply:
cc @marecar3 don't forget to handle the case for self-hosted sites on the WPAndroid also. I can give you login info to my self hosted site if you need for testing.

Reply:
Thanks for the heads up @pinarol!